src/feature/client/entrynodes.c

```diff
@@ -3042,6 +3073,26 @@ entry_guard_parse_from_state(const char *s)
     }
   }
 
+  if (sampled_idx) {
+    int ok = 1;
+    long idx = tor_parse_long(sampled_idx, 10, 0, INT_MAX, &ok, NULL);
+    if (!ok) {
+      log_warn(LD_GUARD, "Guard has invalid sampled_idx %s",
+          escaped(sampled_idx));
+    } else {
+      guard->sampled_idx = (int)idx;
+    }
+  } else if (confirmed_idx) {
+    /* This state has been written by an older Tor version which did not have
+     * sample ordering  */
+
+    guard->sampled_idx = guard->confirmed_idx;
+  } else {
+    log_warn(LD_GUARD, "The state file seems to be into a status that could"
+        " yield to weird entry node selection. It might be better to delete"
+        " it");
```

Comment:
Let's not suggest that people delete their state files.  Instead let's warn, and set the sampled_idx to something.  Leaving sampled_idx at zero seems like a worst-case idea, since it makes the guard get selected before any others.

Reply:
idem as above